server/src/main/java/org/elasticsearch/action/admin/cluster/node/stats/TransportNodesStatsAction.java

```diff
@@ -61,13 +68,42 @@ public TransportNodesStatsAction(
             threadPool.executor(ThreadPool.Names.MANAGEMENT)
         );
         this.nodeService = nodeService;
+        this.client = client;
     }
 
     @Override
     protected NodesStatsResponse newResponse(NodesStatsRequest request, List<NodeStats> responses, List<FailedNodeException> failures) {
         return new NodesStatsResponse(clusterService.getClusterName(), responses, failures);
     }
 
+    @Override
+    protected void newResponseAsync(
+        Task task,
+        NodesStatsRequest request,
+        List<NodeStats> responses,
+        List<FailedNodeException> failures,
+        ActionListener<NodesStatsResponse> listener
+    ) {
+        Set<String> metrics = request.getNodesStatsRequestParameters().requestedMetrics();
+        if (NodesStatsRequestParameters.Metric.ALLOCATIONS.containedIn(metrics)) {
+            client.execute(
+                TransportGetAllocationStatsAction.TYPE,
+                new TransportGetAllocationStatsAction.Request(),
```

Comment:
We need to set the parent task ID on this request so that the `TransportGetAllocationStatsAction` appears in APM as a child of the `TransportNodesStatsAction`